File: src/main/java/com/dumbdogdiner/warrior/gui/DeathSoundGUI.java
package com.dumbdogdiner.warrior.gui;

import com.dumbdogdiner.warrior.Warrior;
import com.dumbdogdiner.warrior.api.WarriorUser;
import com.dumbdogdiner.warrior.api.kit.effects.DeathSound;
import com.dumbdogdiner.warrior.api.kit.effects.DeathSounds;
import com.dumbdogdiner.warrior.api.util.ItemBuilder;
import com.dumbdogdiner.warrior.managers.PlayerManager;
import com.dumbdogdiner.stickyapi.bukkit.gui.ClickableSlot;
import com.dumbdogdiner.stickyapi.bukkit.gui.GUI;

import org.bukkit.Material;
import org.bukkit.event.inventory.InventoryOpenEvent;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public class DeathSoundGUI extends GUI {

    public DeathSoundGUI() {
        super(5, "Deathsounds", Warrior.getInstance());
    }

    @Override
    protected void onInventoryOpen(@NotNull InventoryOpenEvent e) {
        ItemStack placeholder = new ItemBuilder(Material.BLACK_STAINED_GLASS_PANE)
                                .setName(" ").build();

        ItemStack blank = new ItemBuilder(Material.GRAY_DYE)
                          .setName("§7Sound §3???").build();

        for(int i = 0; i < 9; i++) {
            addSlot(i, 0, placeholder);
            addSlot(i, 4, placeholder);
        }

        ClickableSlot exit = new ClickableSlot(Material.BARRIER, 1, "§c§lClose Menu", 4, 4);
        addSlot(exit);

        int soundCount = DeathSound.values().length;
        int i = 0;

        WarriorUser user = PlayerManager.get(e.getPlayer().getUniqueId());

        for(int y = 1; y < 4; y++) {
            for(int x = 1; x < 8; x++) {
                if(i >= soundCount) {
                    addSlot(x, y, blank);
                    continue;
                }

                DeathSound sound = DeathSound.values()[i];

                ItemStack soundItem = new ItemBuilder(Material.LIME_DYE)
                                    .setName("§7Sound §3" + sound.friendlyName())
                                    .build();

                if(DeathSounds.canUse(user, sound)) {
                    ClickableSlot cs = new ClickableSlot(soundItem, x, y);
                    addSlot(cs);
                } else addSlot(x, y, blank);

                i++;
            }
        }
    }
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/gui/DeathSoundGUI.kt
package com.dumbdogdiner.warrior.gui

import com.dumbdogdiner.stickyapi.bukkit.gui.GUI
import com.dumbdogdiner.warrior.Warrior
import com.dumbdogdiner.warrior.api.builders.ItemBuilder
import com.dumbdogdiner.stickyapi.bukkit.gui.ClickableSlot
import com.dumbdogdiner.warrior.api.user.cosmetics.DeathSound
import com.dumbdogdiner.warrior.user.UserCache
import com.dumbdogdiner.warrior.api.user.cosmetics.DeathSounds
import com.dumbdogdiner.warrior.user.User
import org.bukkit.Material
import org.bukkit.event.inventory.InventoryOpenEvent

class DeathSoundGUI : GUI(5, "Deathsounds", Warrior.instance) {

    override fun onInventoryOpen(e: InventoryOpenEvent) {
        val placeholder = ItemBuilder(Material.BLACK_STAINED_GLASS_PANE).setName(" ").build()
        val blank = ItemBuilder(Material.GRAY_DYE).setName("§7Arena §3???").build()

        for (i in 0..8) {
            addSlot(i, 0, placeholder)
            addSlot(i, 4, placeholder)
        }

        val exit = ClickableSlot(Material.BARRIER, 1, "§c§lClose Menu", 4, 4)
        addSlot(exit)

        val user: User = Warrior.userCache[e.player.uniqueId]
        val soundCount = DeathSound.values().size
        var i = 0

        for (y in 1..3) {
            for (x in 1..7) {
                if (i >= soundCount) {
                    addSlot(x, y, blank)
                    continue
                }

                val sound = DeathSound.values()[i]
                val soundItem = ItemBuilder(Material.LIME_DYE)
                    .setName("§7Sound §3" + sound.friendlyName())
                    .build()

                if (DeathSounds.canUse(user, sound)) {
                    val cs = ClickableSlot(soundItem, x, y)
                    addSlot(cs)
                } else addSlot(x, y, blank)

                i++
            }
        }
    }

}